src/System.IO.Compression/src/Interop/Interop.zlib.Windows.cs

```diff
@@ -52,6 +52,23 @@ internal static partial class zlib
             }
         }
 
+        internal static unsafe bool IsCrc32Available()
+        {
+            try
+            {
+                // Make a P/Invoke into zlib crc32 to ensure we're able to find and use it.
+                // If we are, then use zlib.
+                crc32(0, null, 0);
+                return true;
+            }
```

Comment:
This is fine. The IsCrc32Available function isn't used on Desktop since FEATURE_ZLIB_CRC32 is always false.

The whole system around IsCrc32Available and DeflateStream.*.cs WorkerType is wonkier than I'd like, but that's something to solve another day. I'd love to see us switch to an entirely native compression implementation i.e. removal of the managed implementation. Then we could avoid the WorkerType and IsCrc32Available stuff. The only situation where we use it is on a Unix system without Zlib, but in that case I'd rather we pull in a native zlib package from NuGet.